common-utils/src/common/deployment/__init__.py

```diff
@@ -356,125 +378,270 @@ class FlowSpec(BaseModel):
-            cluster=f"prefect-v2-agent-{ENVIRONMENT_TYPE}-{DEPLOYMENT_TYPE}",
-            execution_role_arn=f"arn:aws:iam::{account_id}:role/data-flows-prefect-{DEPLOYMENT_TYPE}-exec-role",
-            task_role_arn=f"arn:aws:iam::{account_id}:role/data-flows-prefect-{DEPLOYMENT_TYPE}-task-role",
-            task_definition={
-                "containerDefinitions": [
-                    {
-                        "secrets": [
-                            {
-                                "name": i.envar_name,
-                                "valueFrom": f"arn:aws:secretsmanager:{AWS_REGION}:{account_id}:secret:{i.secret_name}",
-                            }
-                            for i in self.secrets
-                        ],
-                    }
-                ],
-                "ephemeralStorage": {"sizeInGiB": self.ephemeral_storage_gb},
-            },
+            task_definition_arn=self._handle_task_definition(
+                account_id, ecs_client, slugified_flow_name
+            ),
+            cluster=f"prefect-v2-agent-{DEPLOYMENT_TYPE}",
+            launch_type="FARGATE",
+            allow_task_definition_registration=False,
         )  # type: ignore
         result = ecs_block.save(block_name, overwrite=True)
         LOGGER.info(f"ECS Task Block {block_name} has been pushed with id {result}...")
         return block_name
 
-    def push_deployments(self, flow_path: Path):
+    def push_deployments(
+        self,
+        flow_path: Path,
+        pyproject_metadata: PyProjectMetadata,
+        account_id: str,
+        ecs_client: object,
+    ):
         """Method for processing the flows deployments.
 
         Args:
             so that we can leverage the project's filesystem.
             flow_path (Path): Path object for file name pf  flow file.
+
+        Args:
+            flow_path (Path):  Path object for file name of flow file.
+            pyproject_metadata (PyProjectMetadata): metadata from pyproject.toml
+            account_id (str): AWS account id
+            ecs_client (object): valid ecs client
         """
-        project_name = self._project_name
-        ecs_block_name = self._create_ecs_task_block()
-        s3_block_name = f"{project_name}-{ENVIRONMENT_TYPE}-{DEPLOYMENT_TYPE}"
-        s3 = S3.load(s3_block_name)
-        LOGGER.info(f"Loaded S3 Block: {s3_block_name} for S3 Path: {s3.bucket_path}...")  # type: ignore
-        ecs = ECSTask.load(ecs_block_name)
-        LOGGER.info(f"Using ECS Block: {ecs.name}...")  # type: ignore
-        skip_upload = False
-        flow_folder = get_flow_folder(flow_path)
+
+        # init private attrs that are needed downstream
+        project_name = pyproject_metadata.project_name
+        slugified_fn_name = standard_slugify(self.flow.fn.__name__)
+        # setting flow name to "<subproject name>/<parent directory>/<flow function name>
+        flow_name = f"{project_name}.{get_flow_folder(flow_path)}.{slugified_fn_name}"
```

Comment:
here is where we create the period delimited flow name that is applied on deployment